shorebird_code_push/lib/src/shorebird_code_push_ffi.dart

```diff
@@ -0,0 +1,57 @@
+import 'dart:isolate';
+
+import 'package:shorebird_code_push/src/shorebird_code_push_base.dart';
+import 'package:shorebird_code_push/src/updater.dart';
+
+/// {@template shorebird_code_push}
+/// Get info about your Shorebird code push app.
+/// {@endtemplate}
+class ShorebirdCodePushFfi implements ShorebirdCodePushBase {
+  /// {@macro shorebird_code_push}
+  ShorebirdCodePushFfi({Updater? updater}) : _updater = updater ?? Updater();
+
+  final Updater _updater;
+
+  @override
+  Future<bool> isNewPatchAvailableForDownload() {
+    return Isolate.run(_updater.checkForUpdate);
+  }
+
+  @override
+  Future<int?> currentPatchNumber() {
+    return Isolate.run(() {
+      final currentPatchNumber = _updater.currentPatchNumber();
+      return currentPatchNumber == 0 ? null : currentPatchNumber;
+    });
+  }
+
+  @override
+  Future<int?> nextPatchNumber() {
+    return Isolate.run(
+      () {
+        final patchNumber = _updater.nextPatchNumber();
+        return patchNumber == 0 ? null : patchNumber;
+      },
+    );
+  }
+
+  @override
+  Future<void> downloadUpdateIfAvailable() async {
+    await Isolate.run(_updater.downloadUpdate);
+  }
+
+  @override
+  Future<bool> isNewPatchReadyToInstall() async {
```

Comment:
This is fine, but we also could just add this direct method on the library if it's needed often.

Reply:
The behavior here is also more of telling the caller that the next boot will be a different version.  The patch is already installed at this point.

Reply:
This API was already part of the bindings, I just moved it into a different file/reorganized things a bit.